docs/nddata/covariance.rst

```diff
@@ -0,0 +1,604 @@
+       [0. , 0. , 0. , 0. , 0. , 0. , 0.8, 2. , 4. , 2. ],
+       [0. , 0. , 0. , 0. , 0. , 0. , 0. , 0.8, 2. , 4. ]])
+
+Or likewise for heteroscedastic data:
+
+>>> new_var = (1. + np.absolute(np.arange(npts) - npts//2).astype(float))**2
+>>> new_var
+array([36., 25., 16.,  9.,  4.,  1.,  4.,  9., 16., 25.])
+>>> new_c = c * np.sqrt(new_var[:,None] * new_var[None,:])
+>>> new_c
+array([[36. , 15. ,  4.8,  0. ,  0. ,  0. ,  0. ,  0. ,  0. ,  0. ],
+       [15. , 25. , 10. ,  3. ,  0. ,  0. ,  0. ,  0. ,  0. ,  0. ],
+       [ 4.8, 10. , 16. ,  6. ,  1.6,  0. ,  0. ,  0. ,  0. ,  0. ],
+       [ 0. ,  3. ,  6. ,  9. ,  3. ,  0.6,  0. ,  0. ,  0. ,  0. ],
+       [ 0. ,  0. ,  1.6,  3. ,  4. ,  1. ,  0.8,  0. ,  0. ,  0. ],
+       [ 0. ,  0. ,  0. ,  0.6,  1. ,  1. ,  1. ,  0.6,  0. ,  0. ],
+       [ 0. ,  0. ,  0. ,  0. ,  0.8,  1. ,  4. ,  3. ,  1.6,  0. ],
+       [ 0. ,  0. ,  0. ,  0. ,  0. ,  0.6,  3. ,  9. ,  6. ,  3. ],
+       [ 0. ,  0. ,  0. ,  0. ,  0. ,  0. ,  1.6,  6. , 16. , 10. ],
+       [ 0. ,  0. ,  0. ,  0. ,  0. ,  0. ,  0. ,  3. , 10. , 25. ]])
+
+Reordering and Subsets
+----------------------
+
+When reordering or down-selecting subsets of the elements of :math:`\mathbf{x}`,
+these changes must be propagated to the associated covariance matrix, just as
+would be needed for the error vector for an uncorrelated dataset.
+
+The example below first generates a vector with a shuffled set of indices.  The
+reordered :math:`\mathbf{x}` vector would be constructed by setting
+``reordered_x = x[i]`` and the covariance matrix would be reordered using
+`numpy.ix_`, as follows:
+
+>>> rng = np.random.default_rng(99)
+>>> i = np.arange(npts)
+>>> rng.shuffle(i)
+>>> i
+array([4, 6, 0, 3, 8, 1, 2, 5, 7, 9])
+>>> reordered_c = c[np.ix_(i,i)]
+>>> reordered_c
+array([[1. , 0.2, 0. , 0.5, 0. , 0. , 0.2, 0.5, 0. , 0. ],
+       [0.2, 1. , 0. , 0. , 0.2, 0. , 0. , 0.5, 0.5, 0. ],
+       [0. , 0. , 1. , 0. , 0. , 0.5, 0.2, 0. , 0. , 0. ],
+       [0.5, 0. , 0. , 1. , 0. , 0.2, 0.5, 0.2, 0. , 0. ],
+       [0. , 0.2, 0. , 0. , 1. , 0. , 0. , 0. , 0.5, 0.5],
+       [0. , 0. , 0.5, 0.2, 0. , 1. , 0.5, 0. , 0. , 0. ],
+       [0.2, 0. , 0.2, 0.5, 0. , 0.5, 1. , 0. , 0. , 0. ],
+       [0.5, 0.5, 0. , 0.2, 0. , 0. , 0. , 1. , 0.2, 0. ],
+       [0. , 0.5, 0. , 0. , 0.5, 0. , 0. , 0.2, 1. , 0.2],
+       [0. , 0. , 0. , 0. , 0.5, 0. , 0. , 0. , 0.2, 1. ]])
+
+Note that the diagonal of ``reordered_c`` is still unity (all elements of
+:math:`\mathbf{x}` are perfectly correlated with themselves), but the
+off-diagonal terms have been rearranged to maintain the pre-existing
+correlations.
+
+Creating a covariance matrix for a subset of data is a very similar operation.
+If we want the covariance matrix for the first 3 elements of the data vector, we
+can do the following:
+
+>>> i = np.arange(3)
+>>> sub_c = c[np.ix_(i,i)]
+>>> sub_c
+array([[1. , 0.5, 0.2],
+       [0.5, 1. , 0.5],
+       [0.2, 0.5, 1. ]])
+
+In N-dimensions
```

Comment:
Same as the previous section, find a place to mention that `Covariance` provides methods for these.

Reply:
Note added.